toolbox/siesta/barriers/interstitial.py

```diff
@@ -0,0 +1,206 @@
+           trace_atom_final_position = final Atom Position/Index
+           interpolation_method = idpp
+    """
+    
+    def __init__(self,
+                 host_structure ,
+                 number_of_images  ,
+                 trace_atom_initial_position ,
+                 trace_atom_final_position ,
+                 interpolation_method = 'idpp',
+                 ghost = True ,
+                 ):
+   
+        super().__init__(
+                 neb_scheme = 'interstitial',
+                 relaxed = False,
+                 host_structure = host_structure ,
+                 number_of_images = number_of_images,
+                 initial_relaxed_path = None,
+                 initial_relaxed_fdf_name = None ,
+                 final_relaxed_path = None ,
+                 final_relaxed_fdf_name = None,
+                 trace_atom_initial_position = trace_atom_initial_position ,
+                 trace_atom_final_position = trace_atom_final_position ,
+                 atol = 1e-2,
+                 rtol = 1e-2,
+                )
+        self.host_structure = host_structure
+        self.number_of_images = number_of_images
+        self.interpolation_method = interpolation_method 
+        self.ghost = ghost
+
+    #---------------------------------------------------------
+    # Main Methods
+    #---------------------------------------------------------
+ 
+    def Generate_Interstitial_Images(self):
+        """
+
+        """
+        from .Utils.utils_siesta import read_siesta_fdf,read_siesta_XV ,read_siesta_XV_before_relax, FixingSislImages
+        from .Utils.utils_interstitial import pre_prepare_sisl ,is_frac_or_cart_or_index,pre_prepare_ase_after_relax
+        import os,sys 
+        import glob,shutil
+        from .BarriersIO import SiestaBarriersIO
+
+
+        import sisl
+        from ase.neb import NEB 
+        if self.relaxed == True:
+             print ("=================================================")
+             print ("The Relaxed Interstitial Image Generation ...")
+             print ("=================================================")
+
+             if self.initial_relaxed_path == None or self.final_relaxed_path == None :
+                 sys.exit("intial/final relaxed path not provided")
+             if self.initial_relaxed_fdf_name == None or self.final_relaxed_fdf_name == None :
+                 sys.exit("intial/final relaxed fdf not provided")
+
+             #self.host_structure = read_siesta_fdf(self.host_path,self.host_fdf_name)
+             self.initial_structure = read_siesta_XV(self.initial_relaxed_path,self.initial_relaxed_fdf_name)
+             self.final_structure = read_siesta_XV(self.final_relaxed_path,self.final_relaxed_fdf_name)
+             
+             self.check_ghost()
+             self.test = pre_prepare_ase_after_relax(self.initial_structure['XV'],self.final_structure['XV'],self.ghost)
+             initial = sisl.Geometry.toASE(self.test['initial'])
```

Comment:
sisl now prefers `Geometry.to.ase()` instead, you should have experienced tons of warnings ;)